R/write_pxl_file.R

```diff
@@ -663,14 +682,14 @@ WriteMPX_pxl_file <- function (
       return(cg@counts %>% Matrix::t())
     }) %>% do.call(cbind, .)
     colnames(merged_counts) <- 1:ncol(merged_counts)
-    #merged_counts <- SeuratObject::RowMergeSparseMatrices(all_counts[[1]], all_counts[-1])
 
     merged_layouts <- lapply(names(layouts), function(layout_type) {
       layout_tbl <- layouts[[layout_type]]
+      # If the layout only has x,y columns, add a third z column of NAs
       if (ncol(layout_tbl) == 2 & !"z" %in% names(layout_tbl)) {
         layout_tbl <- layout_tbl %>% mutate(z = NA_real_)
       }
-      layout_tbl <- layout_tbl[, c("x", "y", "z")]
+      layout_tbl <- layout_tbl %>% select(all_of(c("name", "x", "y", "z")))
```

Comment:
Could use `any_of` instead to account for when z is missing.

Reply:
See my answer above. I don't think that will match how the layouts should be stored in the PXL files.